Casks/e/endnote.rb

```diff
@@ -1,38 +1,66 @@
 cask "endnote" do
-  version "21.5.0.20846"
+  # The second part of the `version` is the product version (e.g. X9, 2025).
+  # Until we can identify this using livecheck, we will need to manually find
+  # the new product version on each major version update. See:
+  # https://support.clarivate.com/Endnote/s/article/Download-EndNote
+  version "22.0.0.0,2025"
   sha256 :no_check # required as upstream package is updated in-place
 
-  url "https://download.endnote.com/downloads/#{version.major}/EndNote#{version.major}Installer.dmg"
+  url "https://download.endnote.com/downloads/#{version.csv.second}/EndNote#{version.csv.second}Installer.dmg"
   name "EndNote"
   desc "Reference manager"
   homepage "https://endnote.com/"
 
+  # The XML file doesn't provide any update information for new major versions
+  # (e.g. 22.0.0.0) until a subsequent version is released. The check will fail
+  # when the cask is updated to a new major version but it should automatically
+  # work again when the next version is released (i.e. livecheck will need to
+  # be skipped on CI for major version updates).
   livecheck do
-    url "https://download.endnote.com/updates/#{version.major}.0/EN#{version.major}MacUpdates.xml"
+    url "https://download.endnote.com/updates/#{version.csv.first.major}.0/EN#{version.csv.first.major}MacUpdates.xml"
     regex(/^v?(\d+(?:\.\d+)+)$/i)
     strategy :xml do |xml, regex|
-      xml.get_elements("//updates/build").map do |item|
-        next if item.elements["type"]&.text&.include?("html")
+      current_major = version.csv.first.major
+      current_product_version = version.csv.second
 
-        version = item.elements["updateTo"].text&.strip
-        match = version.match(regex) if version
+      xml.get_elements("//updates/build").map do |item|
+        update_to = item.elements["updateTo"].text&.strip
```

Comment:
```suggestion
        update_to = item.elements["updateTo"]&.text&.strip
```

We should probably use the safe navigation operator here, since the returned element can be `nil`.

Reply:
Agreed, thanks for the catch. I pushed a commit to address this.